Improve `Beatmap_ranking_procedure` and `Beatmap_ranking_procedure/Ranking_queue` (#13991)
ppy/osu-wiki: wiki/Beatmap_ranking_procedure/en.md

## Qualification

Qualification is a process where beatmaps that receive enough nominations are moved to the [Qualified](/wiki/Beatmap/Category#qualified) beatmap category. Beatmaps with only one primary [game mode](/wiki/Game_mode) among all [difficulties](/wiki/Beatmap#difficulty) need two nominations for the primary game mode, while hybrid sets[^hybrid-sets] need two nominations for the main game mode[^main-game-mode] and one for each subsidiary one.

The [Qualified](/wiki/Beatmap/Category#qualified) category exists to provide beatmaps wider exposure to the osu! community with the hopes of discovering potential problems. This is the most common stage for beatmaps to receive wide support, feedback and sometimes meet [obstructions](#obstruction).

> ## Obstruction

Author: I understand that and cant agree more. And I hesitated last month because obstruction is not a very clear word in this situation to convey a good faith for mappers reading this article...

How can I improve this? I mean I still want some clarity for orders between these procedures (like feedback -> nom -> qua -> rank) here. How can I make this smooth as DQs (aka nom resets), Vetoes, and Unranks may happen?

Kinda hope for some suggestions on this. 👀

> ## Obstruction

Reviewer: Think the current flow without this section should be clear enough. At most, a separate segment that covers `Quality assurance` as a general term for feedback posted post-Qualified could be good and would offer some information. 

Just by changing the name of this section and expanding on the currently written text to go a bit deeper into how community members can contribute to the Ranked process would do well. That can then be tied to an off-shot comment going over how maps may be disqualified to iteratively improve upon receiving community feedback, or something along those lines.

Regardless, the word "obstruction" by itself should be scraped.

Nomination resets allow the host to make changes to the beatmap before requesting a re-nomination. It also helps to ensure that modders, BNs, and NAT members check over the most up-to-date version of a beatmap before it enters the [ranking queue](Ranking_queue).

### Vetoes

A [beatmap veto](/wiki/People/Beatmap_Nominators/Beatmap_Veto) allows a BN or NAT member to withhold a beatmap from the [Ranked](/wiki/Beatmap/Category#ranked) section if they feel there are significant issues with its quality. It allows seeking further discussion or clarification over any kind of subjective issues which need to be addressed before the map can proceed with [qualification](#qualification).

## Rank

> When a beatmap stays in Qualified for at least 7 days without any open [problems or suggestions](/wiki/Modding#types-of-mod-posts), the [ranking queue](Ranking_queue) is able to move it to [Ranked](/wiki/Beatmap/Category#ranked). If it has been re-qualified after a disqualification, the time for it to move from Qualified to Ranked may be [recalculated](Ranking_queue#dq-and-re-qualification). Ranked beatmaps have [score leaderboards](/wiki/Ranking) and reward players with performance points.

Reviewer 2: ```suggestion
When a beatmap stays in Qualified for at least 7 days without any open [problems or suggestions](/wiki/Modding#types-of-mod-posts), the [ranking queue](Ranking_queue) is able to move it to [Ranked](/wiki/Beatmap/Category#ranked). If it has been re-qualified after a disqualification, the time for it to move from Qualified to Ranked may be [recalculated](Ranking_queue#dq-and-re-qualification). Ranked beatmaps have [score leaderboards](/wiki/Ranking) and reward players with [performance points](/wiki/Performance_points).
```

Ranked beatmaps are only unranked under exceptional circumstances when issues are found shortly after they reach Ranked status.

## Notes

> [^main-game-mode]: A beatmap's game mode is considered its "main" if it contains the most difficulties. In the event the amount of difficulties is the same between 2+ modes, the main game mode will be the one with the most difficulties mapped by the beatmap's host. If neither of those conditions are met, then the main game mode will be the one which gets nominated first.

Reviewer 2: This PR could be a good opportunity to also update the main game mode definition. Based on feedback I observed in the past, this block of text is too dense and can be difficult to understand.

The initial announcement formatted this definition in a list, which made things much easier to understand. Here's the relevant part of the announcement for reference:

```md
This introduces the concept of a "main game mode", which would need 2 nominations, with the rest of the game modes of a set only needing 1. The main game mode is defined as:

- The game mode with the most difficulties.
- If the amount of difficulties is the same between modes, the main game mode is instead the game mode which the beatmapset host mapped the most difficulties of.
- If the previous two rules do not produce a result, the main game mode is any game mode meeting the previous two criteria.
```

I think we should do something similar to this and have it be its own header, rather than one condensed tooltip.

Author: I agree. By the way, is this concept "main gaime mode" only used here? I pretty hesitate on putting a "titled" section of main game mode **here** (it doesn't seem like it should belongs to `ranking_procedure` I think...) Maybe there's someplace about describing "beatmapsets" and their relationship to single difficulties under the setID, that we can put this concept in?

Reviewer 2: Main game mode is only relevant to the ranking procedure actually, as it's there to determine the required nominations for a beatmap to get qualified.

`ranking_procedure` would be the primary place for it.

Author: OK! So should this be a ## title at the bottom of the article (above tooltips) or a ### one under `## Qualification` (siblinging to Nomination resets and Vetoes looks kind of weird because they are not describing things at a same level)
Personally I don't clearly know where should I put something like this:

```
### Determining the main game mode

For hybrid sets, the **main game mode** is determined by the following order of priority:

1. The game mode containing the most difficulties in the beatmapset.
2. If two or more modes have the same amount of difficulties, the main game mode is instead the game mode which the beatmapset host mapped the most difficulties of.
3. If the previous two rules do not produce a result, the main game mode is any game mode meeting the previous two criteria.
```

Also, rule 3 in the list you show seems not matching the previous tooltip under this article, which I mean, "any game mode" and "the first game mode that gets nominated". Is that an issue?

Author: Im about to accept putting this part above `### Nomination resets` as the first subsection under `Qualification`. Not that weird to sibling the other two. Does that sound good or..

Reviewer 2: > Im about to accept putting this part above `### Nomination resets` as the first subsection under `Qualification`. Not that weird to sibling the other two. Does that sound good or..

Sounds good.

> Also, rule 3 in the list you show seems not matching the previous tooltip under this article, which I mean, "any game mode" and "the first game mode that gets nominated". Is that an issue?

Wording should match what's in the tooltip, as that's the most up-to-date rendition of the logic. Go with "the first game mode that gets nominated"

Author: okoko then. Changes will be added in next commit :')

## Qualification

> Qualification is a process where beatmaps that receive enough nominations are moved to the [Qualified](/wiki/Beatmap/Category#qualified) beatmap category. Beatmaps with only one primary [game mode](/wiki/Game_mode) among all [difficulties](/wiki/Beatmap#difficulty) only need two nominations, while hybrid sets[^hybrid-sets] need two nominations for the **main game mode** (talked about below) and one for each subsidiary one.

Author: All should be solved now =u

Qualified beatmaps receive wider exposure to the osu! community as a final **quality assurance** step before becoming permanently [Ranked](/wiki/Beatmap/Category#ranked). This is the most common stage for beatmaps to receive wide support, feedback, and undergo improvements if issues are found. During this period:

- Community members can test or mod a beatmap, and then provide feedback and report issues[^report-correctly] through the beatmap's discussion page.
- The BN and NAT teams are automatically notified of all suggestions and problems. Users who have enabled notification for new problems on qualified beatmaps will also be notified.
- Suggestions and problems can be discussed and resolved collaboratively between the community and the beatmap's creator.
- Qualified beatmaps cannot be directly updated by its creators.

When issues are identified that require changes, the beatmap may undergo a [nomination reset](#nomination-resets), allowing the creator to address feedback while maintaining the integrity of the ranking procedure.

> ### Determining the main game mode

Reviewer 2: ```suggestion
### Determining the main game mode {id=main-mode}
```

For hybrid sets, the **main game mode** is determined by the following order of priority:

1. The game mode containing the most difficulties in the beatmapset.
2. If two or more modes have the same amount of difficulties, the main game mode is instead the game mode which the beatmapset host mapped the most difficulties of.

> 3. If the previous two rules do not produce a result, the main game mode is any game mode meeting the previous two criteria.

Reviewer 2: ```suggestion
3. If the previous two rules do not produce a result, the main game mode will be the one which gets nominated first.
```

Author: oops my bad 😢
